src/messages.js

```diff
@@ -65,9 +72,111 @@
   JSAV.ext.clearumsg = function(msg, options) {
     this._msg.clear();
   };
-  
+
+  ////////////////////////
+  ///////TextToSpeech/////
+  JSAV.ext.textToSpeech = function(speechText) {
+    var modifiedMsg;
+    modifiedMsg = speechText.replace(/<[^>]*>/g, "");
+    modifiedMsg = modifiedMsg.replace(/\$/g, "");
+    modifiedMsg = modifiedMsg.replace(/\\mathbf/gi, "");
+    modifiedMsg = modifiedMsg.replace(/\\displaystyle/gi, "");
+    modifiedMsg = modifiedMsg.replace(/\\mbox/gi, "");
+    modifiedMsg = modifiedMsg.replace(/n-/gi, "n minus");
+    modifiedMsg = modifiedMsg.replace(/m-/gi, "m minus");
+    modifiedMsg = modifiedMsg.replace(/%/gi, "remainder");
+    var synth = window.speechSynthesis;
+    var u = new SpeechSynthesisUtterance();
+    var amISpeaking = synth.speaking;
+    u.lang = "en-US";
+    u.rate = 1.0;
+    // Assign the umsg text to the narration file
+    u.text = modifiedMsg;
+    if(amISpeaking === true) {
+      synth.cancel();
+    }
+    synth.speak(u);
+  };
+  //////////////////////////////////////////
+////////////Sound control button
+  JSAV.ext.soundSettings = function(jsav) {
```

Comment:
There shouldn't be a need to pass the `jsav` object as a parameter. `soundSettings`will be an instance function of JSAV objects, and you're already using `this` to refer to the JSAV instance.